api/src/experimental/metrics/ExperimentalMeter.ts

```diff
@@ -0,0 +1,129 @@
+ *     - failure to do this will result in failing builds for users that update the SDK but not the API
+ * - add the new functionality to the {@link IExperimentalMeter} interface
+ * - implement the functionality in the {@link ExperimentalMeter}
+ *   - if the underlying {@link Meter} used as an {@link IExperimentalMeter} throws an error, return a No-Op implementation.
+ *
+ * Users may now use {@link wrapMeter} to get access to experimental SDK functionality via the API
+ *
+ * To stabilize a feature:
+ * - move the function interface from `IExperimentalMeter` to `Meter`
+ * - replace the implemented method with a property of the same type as the one from `Meter`, move any auxiliary types to stable
+ * - In ExperimentalMeter
+ *   - remove the implementation that falls back to No-Op on errors.
+ *   - assign the function that was moved to `Meter` to the property defined in the last step
+ * - Implement the no-op case in {@link NoopMeter}
+ * - ensure any `@experimental` annotations are removed
+ */
+
+/**
+ * @experimental
+ *
+ * Meter that offers experimental functionality IF that functionality is implemented by the SDK.
+ * MAY return a no-op otherwise. Stable features continue to work as expected.
+ */
+export interface IExperimentalMeter extends Meter {
+  /**
+   * @experimental Will be added to {@link Meter} in a future version when the specification is marked stable
+   *
+   * Creates and returns a new `Gauge`.
+   * @param name the name of the metric.
+   * @param [options] the metric options.
+   */
+  createGauge<AttributesTypes extends MetricAttributes = MetricAttributes>(
+    name: string,
+    options?: MetricOptions
+  ): Gauge<AttributesTypes>;
+}
+
+class ExperimentalMeter implements IExperimentalMeter {
+  private _meter: Meter;
+  constructor(meter: Meter) {
+    this._meter = meter;
+    this.addBatchObservableCallback =
+      meter.addBatchObservableCallback.bind(meter);
+    this.createCounter = meter.createCounter.bind(meter);
+    this.createObservableGauge = meter.createObservableGauge.bind(meter);
+    this.createHistogram = meter.createHistogram.bind(meter);
+    this.createObservableCounter = meter.createObservableCounter.bind(meter);
+    this.createObservableUpDownCounter =
+      meter.createObservableUpDownCounter.bind(meter);
+    this.createUpDownCounter = meter.createUpDownCounter.bind(meter);
+    this.removeBatchObservableCallback =
+      meter.removeBatchObservableCallback.bind(meter);
+  }
+  addBatchObservableCallback: Meter['addBatchObservableCallback'];
+  createCounter: Meter['createCounter'];
+  createObservableGauge: Meter['createObservableGauge'];
+  createHistogram: Meter['createHistogram'];
+  createObservableCounter: Meter['createObservableCounter'];
+  createUpDownCounter: Meter['createUpDownCounter'];
+  createObservableUpDownCounter: Meter['createObservableUpDownCounter'];
+  removeBatchObservableCallback: Meter['removeBatchObservableCallback'];
```

Comment:
This seems like a lot of duplication and large strings. I doubt this is very friendly to minification. Isn't there a way to keep the meter mostly the same and put new methods on it without a whole new class?

Reply:
The strings actually don't end up in the final bundle as they're only used for the types. The rest of the code, however is not the most efficient in terms of bundle-size, you're right.

Since the API we were trying to add here was marked as stable, I'll close this PR and re-focus my efforts on #4669